vavr/src/main/java/org/jdbi/v3/vavr/VavrGenericMapUtil.java

```diff
@@ -57,20 +56,13 @@ static Type resolveMultimapEntryType(Type mapType) {
         return resolveMaplikeEntryType(mapType, KEY_MULTIMAP, VALUE_MULTIMAP);
     }
 
-    private static Type resolveMaplikeEntryType(Type maplikeType, Type keyParam, Type valueParam) {
-        Type keyType = GenericTypes.resolveType(keyParam, maplikeType);
-        Type valueType = GenericTypes.resolveType(valueParam, maplikeType);
+    private static Type resolveMaplikeEntryType(Type maplikeType, TypeVariable<? extends Class<?>> keyParam, TypeVariable<? extends Class<?>> valueParam) {
+        Type keyType = GenericTypeReflector.getTypeParameter(maplikeType, keyParam);
+        Type valueType = GenericTypeReflector.getTypeParameter(maplikeType, valueParam);
         return resolveMapEntryType(keyType, valueType);
     }
 
     private static Type resolveMapEntryType(Type keyType, Type valueType) {
-        return resolveMapEntryType(TypeToken.of(keyType), TypeToken.of(valueType));
-    }
-
-    private static <K, V> Type resolveMapEntryType(TypeToken<K> keyType, TypeToken<V> valueType) {
-        return new TypeToken<Tuple2<K, V>>() {}
-                .where(new TypeParameter<K>() {}, keyType)
-                .where(new TypeParameter<V>() {}, valueType)
-                .getType();
+        return TypeFactory.parameterizedClass(Tuple2.class, keyType, valueType);
```

Comment:
This is another argument in favor of re-exposing this method through e.g. `GenericType.parameterized(Class, Type...)`

Reply:
In fact this whole class could go away.

Reply:
Nice